### ISSUE TITLE
sound: add functionality for pausing/stopping sound

### ISSUE THREAD

**Issue by JaStarbuck** [uchicago-cs/chiventure#1464]
Summary: As seen in the demo.c file, they include functionality past load/free/play sound. Specifically, it looks like they are able to introduce sound effects, stop music, pause music, etc from the press of a button or through other inputs. In this issue, the goal is to add functions in the sound module that specifically allow the pausing/stopping of music or sound effects. It looks like there are built in SDL functions in the demo that will be helpful in the process, and we will need to decide how and where to implement button pressing in order to pause/stop the sounds. We hope that this will further enhance the chiventure experience and will make it easier and more appealing for other groups to use the sound module in their own projects.

assignees: Jason, Victor, Rhea

**Comment by JaStarbuck** [comment]
Now that the fixing sound module warnings issue has been closed and merged into dev, we can seriously begin work on this issue.

**Comment by JaStarbuck** [comment]
I have created a new branch called sound/pause_stop that will contain the work for these modules.

**Comment by JaStarbuck** [comment]
Looking through the demo.c code, it is important to know what an event is. I found this website to be useful: https://www.libsdl.org/release/SDL-1.2.15/docs/html/sdlevent.html

**Comment by JaStarbuck** [comment]
Also, this website!!!:
https://www.libsdl.org/projects/SDL_mixer/docs/SDL_mixer_52.html
this shows a ton of functions that can increase functionality in the sound module.

**Comment by JaStarbuck** [comment]
I've realized while working that including free_sound in the play_sound function doesn't make sense--it will be up to the chiventure developer to use free_sound I believe. I have thus edited the play_sound function for that reason. I have also found it useful to include a resume_sound function that complements the pause_sound function.

**Comment by JaStarbuck** [comment]
Closing statement: The branch for this issue has been merged into the dev branch. In this issue, we have created the function headers in sound.h and created the simple functions for pause/stop/resume_sound in sound.c. This proved to be very simple functions so, hopefully, won't be too hard to test in the future. It will be exciting to see how other groups and the future sound group will implement these functions to help develop chiventure. Hopefully, these functions will supplement the already existing play_sound function. For the future, it'd also be interesting to see if using these functions in reaction to key presses could be implemented.
